Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift

```diff
@@ -0,0 +1,200 @@
+                  typealias Output = Echo_EchoResponse
+                  static let descriptor = MethodDescriptor(
+                      service: echo.Echo,
+                      method: Collect
+                  )
+              }
+              enum Expand {
+                  typealias Input = Echo_EchoRequest
+                  typealias Output = Echo_EchoResponse
+                  static let descriptor = MethodDescriptor(
+                      service: echo.Echo,
+                      method: Expand
+                  )
+              }
+              enum Get {
+                  typealias Input = Echo_EchoRequest
+                  typealias Output = Echo_EchoResponse
+                  static let descriptor = MethodDescriptor(
+                      service: echo.Echo,
+                      method: Get
+                  )
+              }
+              enum Update {
+                  typealias Input = Echo_EchoRequest
+                  typealias Output = Echo_EchoResponse
+                  static let descriptor = MethodDescriptor(
+                      service: echo.Echo,
+                      method: Update
+                  )
+              }
+              static let methods: [MethodDescriptor] = [
+                  echo.Echo.Collect.descriptor,
+                  echo.Echo.Expand.descriptor,
+                  echo.Echo.Get.descriptor,
+                  echo.Echo.Update.descriptor
+              ]
+              typealias StreamingServiceProtocol = echo_EchoServiceStreamingProtocol
+              typealias ServiceProtocol = echo_EchoServiceProtocol
+          }
+      }
+      """
+    try self._assertTypealiasTranslation(
+      codeGenerationRequest: self.codeGenerationRequest,
+      expectedSwift: expectedSwift
+    )
+  }
+}
+
+extension SnippetBasedTranslatorTests {
+  private func diff(expected: String, actual: String) throws -> String {
+    let process = Process()
+    process.executableURL = URL(fileURLWithPath: "/usr/bin/env")
+    process.arguments = [
+      "bash", "-c",
+      "diff -U5 --label=expected <(echo '\(expected)') --label=actual <(echo '\(actual)')",
+    ]
+    let pipe = Pipe()
+    process.standardOutput = pipe
+    try process.run()
+    process.waitUntilExit()
+    let pipeData = try XCTUnwrap(
+      pipe.fileHandleForReading.readToEnd(),
+      """
+      No output from command:
+      \(process.executableURL!.path) \(process.arguments!.joined(separator: " "))
+      """
+    )
+    return String(decoding: pipeData, as: UTF8.self)
+  }
+
+  private func XCTAssertEqualWithDiff(
+    _ actual: String,
+    _ expected: String,
+    file: StaticString = #filePath,
+    line: UInt = #line
+  ) throws {
+    if actual == expected { return }
+    XCTFail(
+      """
+      XCTAssertEqualWithDiff failed (click for diff)
+      \(try diff(expected: expected, actual: actual))
+      """,
+      file: file,
+      line: line
+    )
+  }
+
+  func _assertTypealiasTranslation(
```

Comment:
This shouldn't need to be prefixed with an underscore. It can also be `private`.